src/main/java/com/minecolonies/coremod/items/ItemSupplyCampDeployer.java

```diff
@@ -0,0 +1,211 @@
+     */
+    private static final int CHECK_X_DIRECTIONS = 4;
+
+    /**
+     * Creates a new supplychest deployer. The item is not stackable.
+     */
+    public ItemSupplyCampDeployer()
+    {
+        super("supplyCampDeployer");
+
+        super.setCreativeTab(ModCreativeTabs.MINECOLONIES);
+        setMaxStackSize(1);
+    }
+
+    @Override
+    public EnumActionResult onItemUse(
+            final EntityPlayer player,
+            final World worldIn,
+            final BlockPos pos,
+            final EnumHand hand,
+            final EnumFacing facing,
+            final float hitX,
+            final float hitY,
+            final float hitZ)
+    {
+        if(worldIn == null || player == null)
+        {
+            return EnumActionResult.FAIL;
+        }
+
+        final ItemStack stack = player.getHeldItem(hand);
+        if (worldIn.isRemote || stack.getCount() == 0 || !isFirstPlacing(player))
+        {
+            return EnumActionResult.FAIL;
+        }
+
+        final EnumFacing dir = player.getHorizontalFacing();
+        if (spawnCamp(worldIn, pos, dir))
+        {
+            worldIn.setBlockState(pos.up(), Blocks.CHEST.getDefaultState().withProperty(BlockChest.FACING, dir));
+
+            fillChest((TileEntityChest) worldIn.getTileEntity(pos.up()));
+
+            stack.setCount(stack.getCount()-1);
+            player.addStat(ModAchievements.achievementGetSupply);
+
+            return EnumActionResult.SUCCESS;
+        }
+        LanguageHandler.sendPlayerMessage(player, "item.supplyCampDeployer.invalid");
+        return EnumActionResult.FAIL;
+    }
+
+    /**
+     * Checks if the player already placed a supply chest.
+     *
+     * @param player the player.
+     * @return boolean, returns true when player hasn't placed before, or when infinite placing is on.
+     */
+    private static boolean isFirstPlacing(@NotNull final EntityPlayer player)
+    {
+        if (Configurations.allowInfiniteSupplyChests || !player.hasAchievement(ModAchievements.achievementGetSupply))
+        {
+            return true;
+        }
+        LanguageHandler.sendPlayerMessage(player, "com.minecolonies.coremod.error.supplyChestAlreadyPlaced");
+        return false;
+    }
+
+    /**
+     * Spawns the ship and supply chest.
+     *
+     * @param world world obj.
+     * @param pos   coordinate clicked.
+     */
+    private boolean spawnCamp(@NotNull final World world, @NotNull final BlockPos pos, @NotNull final EnumFacing chestFacing)
+    {
+        if (isInsideAColony(world, pos))
+        {
+            return false;
+        }
+        return checkAndPlaceSupplyCamp(world, pos, chestFacing);
+    }
+
+    private boolean checkAndPlaceSupplyCamp(final World world, @NotNull final BlockPos pos, @NotNull final EnumFacing direction)
+    {
+        EnumFacing facing = direction;
+        for (int i = 0; i < CHECK_X_DIRECTIONS; i++)
+        {
+            switch (facing)
+            {
+                case NORTH:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1151)